### PR TITLE
Convert CopyButton component to JSX

### PR DESCRIPTION
**Pull request by whymarrh**
Refs #6291

This PR replaces hyperscript in the `CopyButton` component with JSX.

### CODE REVIEW

**Comment by Gudahtt** [comment]
So, ah... I checked to see if _this_ one was unused too, since `Copyable` turned out to be.

This one _is_ used, but only by `shift-list-item`, which is an alternative to `transaction-list-item` that's only used when a transaction has the key `shapeshift`. ShapeShift was removed _ages_ ago, so unless someone has an ancient transaction in their history, that case can't happen anymore. So this one is _kinda_ transitively unused? :sweat_smile: 

I guess we'll leave it in for now in case it is in someone's history, but maybe we should consider removing it soon :thinking:

File: ui/app/components/ui/copyButton.js [hunk 1]
```diff
@@ -1,66 +1,66 @@
...
-  const props = this.props
-  const state = this.state || {}
+  static defaultProps = {
+    title: null,
+  }
 
-  const value = props.value
-  const copied = state.copied
+  static propTypes = {
+    value: PropTypes.any.isRequired,
```
Comment (Gudahtt) [inline ui/app/components/ui/copyButton.js]:
```suggestion
    value: PropTypes.string.isRequired,
```
The `copy-to-clipboard` library accepts only `string` input.

Comment (Gudahtt) [reply ui/app/components/ui/copyButton.js]: ... after further investigation, the value passed to this component for the `value` prop is always `undefined` :unamused: . So this is broken.

Comment (whymarrh) [reply ui/app/components/ui/copyButton.js]: Wait, uhhh hmm?

Comment (Gudahtt) [reply ui/app/components/ui/copyButton.js]: Here's the only place that prop is passed in: https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/components/app/shift-list-item.js#L77

`props.depositAddress` doesn't exist. It's always undefined.

Comment (whymarrh) [reply ui/app/components/ui/copyButton.js]: Thanks, I hate it

**Review by Gudahtt — APPROVED** [review]
I guess this'll do for now! :man_shrugging:

**Pull request merged**